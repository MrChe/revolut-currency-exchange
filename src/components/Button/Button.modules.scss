.Button {
  position: relative;
  background-color: transparent;
  padding: 8px;
  line-height: 1;
  border: 1px solid black;
  cursor: pointer;
  border-radius: 4px;
  outline: none;
}

.Circle {
  width: 48px;
  height: 48px;
  border-radius: 50%;
}

.Bg {
  &_Type {
    &_White {
      background-color: white;
    }
    &_Black {
      background-color: white;
    }
  }
}


.State {
  &_Disabled {
    background-color: rgba(255,255,255, 0.5);
    cursor: default;
    pointer-events: none;
  }
}
