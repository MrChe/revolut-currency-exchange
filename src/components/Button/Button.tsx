import React from "react";
import cn from "classnames";

import styles from "./Button.modules.scss";

interface IButtonProps {
  children: JSX.Element | string;
  disabled?: boolean;
  onClick?: () => void;
  type?: "circle" | "normal";
  bg?: "transparent" | "white" | "black";
  name: string;
}

export const Button = (props: IButtonProps): JSX.Element => {
  return (
    <button
      aria-label={props.name}
      onClick={props.onClick}
      disabled={props.disabled}
      className={cn({
        [styles.Button]: true,
        [styles.Circle]: props.type === "circle",
        [styles.Bg_Type_White]: props.bg === "white",
        [styles.Bg_Type_Black]: props.bg === "black",
        [styles.State_Disabled]: props.disabled,
      })}
    >
      {props.children}
    </button>
  );
};
