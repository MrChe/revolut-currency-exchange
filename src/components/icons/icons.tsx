import { ReactComponent as AccountSVG } from "../../resources/icons/i-account.svg";
import { ReactComponent as ArrowRightSVG } from "../../resources/icons/i-arrow-right.svg";
import { ReactComponent as CardSVG } from "../../resources/icons/i-card.svg";
import { ReactComponent as ExchangeSVG } from "../../resources/icons/i-exchange.svg";
import { ReactComponent as PlusSVG } from "../../resources/icons/i-plus.svg";
import { ReactComponent as ProfileSVG } from "../../resources/icons/i-profile.svg";
import { ReactComponent as SendSVG } from "../../resources/icons/i-send.svg";
import { ReactComponent as SupportSVG } from "../../resources/icons/i-support.svg";

export const Icon = {
  Account: AccountSVG,
  ArrowRight: ArrowRightSVG,
  Card: CardSVG,
  Exchange: ExchangeSVG,
  Plus: PlusSVG,
  Profile: ProfileSVG,
  Send: SendSVG,
  Support: SupportSVG,
};
