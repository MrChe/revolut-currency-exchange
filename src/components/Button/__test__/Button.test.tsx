import React from "react";
import renderer from "react-test-renderer";
import { Button } from "../Button";
import { Icon } from "../../../components/icons/icons";

describe("Button Component", () => {
  it("Button with text snapshot", () => {
    const tree = renderer.create(<Button name={"test"}>Hello</Button>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Button circle with icon snapshot", () => {
    const tree = renderer
      .create(
        <Button name={"test"} type={"circle"}>
          <Icon.Exchange />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button bg white with icon snapshot", () => {
    const tree = renderer
      .create(
        <Button name={"test"} type={"circle"} bg={"white"}>
          <Icon.Exchange />
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button bg white disabled snapshot", () => {
    const tree = renderer
      .create(
        <Button name={"test"} type={"circle"} bg={"white"}>
          Hello
        </Button>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
