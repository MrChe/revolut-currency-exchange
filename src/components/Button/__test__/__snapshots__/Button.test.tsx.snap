// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button bg white disabled snapshot 1`] = `
<button
  aria-label="test"
  className="Button Circle Bg_Type_White"
>
  Hello
</button>
`;

exports[`Button Component Button bg white with icon snapshot 1`] = `
<button
  aria-label="test"
  className="Button Circle Bg_Type_White"
>
  <test />
</button>
`;

exports[`Button Component Button circle with icon snapshot 1`] = `
<button
  aria-label="test"
  className="Button Circle"
>
  <test />
</button>
`;

exports[`Button Component Button with text snapshot 1`] = `
<button
  aria-label="test"
  className="Button"
>
  Hello
</button>
`;
